src/core/reactor_backend.cc

```diff
@@ -1484,10 +1490,101 @@ class reactor_backend_uring final : public reactor_backend {
         delete pfd;
     }
     virtual future<std::tuple<pollable_fd, socket_address>> accept(pollable_fd_state& listenfd) override {
-        return _r.do_accept(listenfd);
+        if (listenfd.take_speculation(POLLIN | POLLOUT)) {
+            try {
+                listenfd.maybe_no_more_recv();
+            } catch (...) {
+                return current_exception_as_future<std::tuple<pollable_fd, socket_address>>();
+            }
+            socket_address sa;
+            try {
+                auto maybe_fd = listenfd.fd.try_accept(sa, SOCK_CLOEXEC);
+                if (maybe_fd) {
+                    listenfd.speculate_epoll(EPOLLIN);
+                    pollable_fd pfd(std::move(*maybe_fd), pollable_fd::speculation(EPOLLOUT));
+                    return make_ready_future<std::tuple<pollable_fd, socket_address>>(std::move(pfd), std::move(sa));
+                }
+            } catch (...) {
+                return current_exception_as_future<std::tuple<pollable_fd, socket_address>>();
+            }
+        }
+        class accept_completion final : public io_completion {
+            pollable_fd_state& _listenfd;
+            socket_address _sa;
+            promise<std::tuple<pollable_fd, socket_address>> _result;
+        public:
+            accept_completion(pollable_fd_state& listenfd)
+                : _listenfd(listenfd) {}
+            void complete(size_t fd) noexcept final {
+                _listenfd.speculate_epoll(EPOLLIN);
+                pollable_fd pfd(file_desc::from_fd(fd), pollable_fd::speculation(EPOLLOUT));
+                _result.set_value(std::move(pfd), std::move(_sa));
+                delete this;
+            }
+            void set_exception(std::exception_ptr eptr) noexcept final {
+                auto set_ex_and_dispose = defer([eptr, this]() noexcept {
```

Comment:
i don't `std::move(eptr)` here or afterwards, because this is a failure path, and because

> std::exception_ptr is a shared-ownership smart pointer 

according to [c++ reference](https://en.cppreference.com/w/cpp/error/exception_ptr)